app/adapters/ssh-key.js

```diff
@@ -0,0 +1,31 @@
+import AuthAdapter from './auth';
+import buildURLWithPrefixMap from '../utils/build-url-with-prefix-map';
+import Ember from 'ember';
+
+export default AuthAdapter.extend({
+  buildURL: buildURLWithPrefixMap({
+    'users': 'user.id'
+  }),
+
+  // To delete an ssh-key, the adapter should send a DELETE to /ssh_keys/:id,
+  // but it POSTs to /users/:user_id/ssh_keys.
+  // Override deleteRecord to remove the "/users/:user_id" from the
+  // url for a DELETE.
+  deleteRecord: function(store, type, record){
+    var id = Ember.get(record, 'id');
+    var userId = Ember.get(record, 'user.id');
+
+    var url = this.buildURL(type.typeKey, id, record);
+
+    if (userId) {
+      url = url.replace("/users/" + userId, "");
+    }
+
+    return this.ajax(url, "DELETE");
+  },
```

Comment:
I was thinking that maybe `buildURLWithPrefix` could take a map of verbs -> prefixes, so it would be, like: `buildURLWithPrefix({'POST': {'users': 'user.id'}});`. Not sure how difficult that would be but it probably makes some sense, since the paths are definitely verb-dependent. I am pretty sure the GET for ssh_keys is also non-prefixed (like the DELETE).